libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts

```diff
@@ -192,19 +192,37 @@ const createConnections = (
   targetSchemaFlattened: SchemaNodeDictionary,
   functions: FunctionData[]
 ) => {
+  const isLoop: boolean = targetKey.includes(mapNodeParams.for);
+  const isConditional: boolean = targetKey.startsWith(mapNodeParams.if);
   const sourceEndOfFunction = sourceNodeString.indexOf('(');
-  const amendedSourceKey = targetKey.includes(mapNodeParams.for) ? getSourceValueFromLoop(sourceNodeString, targetKey) : sourceNodeString;
+  const amendedSourceKey = isLoop ? getSourceValueFromLoop(sourceNodeString, targetKey) : sourceNodeString;
+  let mockDirectAccessFnKey: string | undefined = undefined;
+  const [daOpenBracketIdx, daClosedBracketIdx] = [amendedSourceKey.indexOf('['), amendedSourceKey.indexOf(']')];
 
-  const sourceNode =
-    sourceEndOfFunction > -1
-      ? findFunctionForFunctionName(amendedSourceKey.substring(0, sourceEndOfFunction), functions)
-      : findNodeForKey(amendedSourceKey, sourceSchema.schemaTreeRoot);
+  // Identify source schema node, or Function(Data) from source key
+  let sourceNode: SchemaNodeExtended | FunctionData | undefined = undefined;
+  if (sourceEndOfFunction >= 0) {
```

Comment:
Callout: Felt ">= 0" ever so slightly better represented the condition being tested than "> -1"

Can revert if any objections